fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneIndexMaintainer.java

```diff
@@ -343,6 +343,8 @@ public CompletableFuture<Void> mergeIndex() {
                 .thenCompose(ignored -> {
                     IndexDeferredMaintenanceControl maintenanceControl = state.store.getIndexDeferredMaintenanceControl();
                     maintenanceControl.setLastStep(IndexDeferredMaintenanceControl.LastStep.MERGE);
+                    // This agilityContext will be used to determine/iterate grouping keys and partitions. The time gap between calls might
```

Comment:
I thought I did this correctly, but I later discovered I did not, so I checked out the main for a few relevant classes, and took a new approach to managing the agility context.